src/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -214,6 +184,49 @@ public static void ObjectTree()
             Assert.Equal(3, child.MyOverflow["MyIntMissingChild"].GetInt32());
         }
 
+        [Fact]
+        public static void ExtensionPropertyDictionaryStringObject()
+        {
+            ClassWithExtensionPropertyAlreadyInstantiated obj = JsonSerializer.Parse<ClassWithExtensionPropertyAlreadyInstantiated>(@"{}");
+            obj.MyOverflow.Add("test", new object());
+            Assert.Equal(@"{""MyOverflow"":{""test"":{}}}", JsonSerializer.ToString(obj));
+        }
+
+        [Fact]
+        public static void ExtensionPropertyDictionaryStringObject_Read()
+        {
+            ClassWithExtensionPropertyAsObject obj = JsonSerializer.Parse<ClassWithExtensionPropertyAsObject>(@"{""MyOverflow"":{""key"":1e-005}}");
+            Assert.IsType<double>(obj.MyOverflow["key"]);
+            Assert.Equal(1e-005, (double)obj.MyOverflow["key"]);
+
+            obj = JsonSerializer.Parse<ClassWithExtensionPropertyAsObject>(@"{""MyOverflow"":{""key"":1.1}}");
```

Comment:
We do not want to "guess" here. We just want to create JsonElement on deserialize.

Reply:
Ok understood thanks